The C header file is not included when an imported type has header pragma

mytests/cheader.h:
```nim
typedef int CType;
```

mytests/modbar.nim:
```nim
type CType* {.importc: "CType", header: "cheader.h".} = int32
```

mytests/testimpc.nim:
```nim
import modbar

var x: CType
```

```console
$ ./bin/nimony c -r mytests/testimpc.nim 
bin/nimsem  --isSystem m nimcache/sysvq0asl.1.nif nimcache/sysvq0asl.2.nif nimcache/sysvq0asl.2.idx.nif
bin/nimsem  m nimcache/mod4quvct.1.nif nimcache/mod4quvct.2.nif nimcache/mod4quvct.2.idx.nif
bin/nimsem  --isMain m nimcache/tesf13wcr1.1.nif nimcache/tesf13wcr1.2.nif nimcache/tesf13wcr1.2.idx.nif
cc -c   -Ivendor/mimalloc/include vendor/mimalloc/src/static.c -o nimcache/static.o
bin/hexer --bits:64 nimcache/tesf13wcr1.2.nif
bin/nifc c --compileOnly --isMain  nimcache/tesf13wcr1.c.nif
cc -c  -Imytests  nimcache/tesf13wcr1.c -o nimcache/tesf13wcr1.o
nimcache/tesf13wcr1.c:406:1: error: unknown type name ‘CType’
  406 | CType x_0_tesf13wcr1;
      | ^~~~~
```

--- a/src/hexer/nifcgen.nim
+++ b/src/hexer/nifcgen.nim
@@ -452,6 +452,11 @@ proc traverseType(c: var EContext; n: var Cursor; flags: set[TypeFlag] = {}) =
         c.dest.addSymUse pool.syms.getOrIncl(pool.strings[n.litId] & ".c"), n.info
         inc n
         skipParRi c, n
+        if n.kind != ParRi and n.pragmaKind == HeaderP:
+          inc n
+          c.headers.incl n.litId
+          inc n
+          skipParRi c, n
         skipParRi c, n
       else:
         takeParRi c, n

--- a/src/nimony/sem.nim
+++ b/src/nimony/sem.nim
@@ -2190,6 +2190,7 @@ type
     bits: int
     hasVarargs: PackedLineInfo
     flags: set[PragmaKind]
+    headerFileTok: PackedToken
 
 proc semPragma(c: var SemContext; n: var Cursor; crucial: var CrucialPragma; kind: SymKind) =
   let hasParRi = n.kind == ParLe # if false, has no arguments
@@ -2253,6 +2254,7 @@ proc semPragma(c: var SemContext; n: var Cursor; crucial: var CrucialPragma; kin
       var name = replaceSubs(pool.strings[tok.litId], info.getFile(), c.g.config)
       name = name.toRelativePath(c.g.config.nifcachePath)
       c.dest[idx] = strToken(pool.strings.getOrIncl(name), tok.info)
+      crucial.headerFileTok = c.dest[idx]
     # Finalize expression
     c.dest.addParRi()
   of AlignP, BitsP:
@@ -4554,12 +4556,19 @@ proc fitTypeToPragmas(c: var SemContext; pragmas: CrucialPragma; typeStart: int)
       let info = typ.info
       endRead(c.dest)
       let kind = if ImportcP in pragmas.flags: ImportcP else: ImportcppP
-      var tokens = [
-        parLeToken(pool.tags.getOrIncl($kind), info),
+      var tokens = @[
+        parLeToken(kind, info),
         strToken(pool.strings.getOrIncl(pragmas.externName), info),
         parRiToken(info)
       ]
-      c.dest.insert fromBuffer(tokens), typeStart+2
+      if HeaderP in pragmas.flags:
+        assert pragmas.headerFileTok.kind == StringLit
+        tokens.add [
+          parLeToken(HeaderP, info),
+          pragmas.headerFileTok,
+          parRiToken(info)
+        ]
+      c.dest.insert tokens, typeStart+2
     else:
       let err = "cannot import type " & typeToString(typ)
       let info = typ.info

--- a/tests/nimony/ffi/cinttypedef.h
+++ b/tests/nimony/ffi/cinttypedef.h
@@ -1,0 +1,1 @@
+typedef int CIntTypeDef;

--- a/tests/nimony/ffi/deps/mimportcinttypedef.nim
+++ b/tests/nimony/ffi/deps/mimportcinttypedef.nim
@@ -1,1 +1,1 @@
-type CIntType* {.importc: "CIntType", header: "cinttype.h".} = int32
+type CIntTypeDef* {.importc: "CIntTypeDef", header: "cinttypedef.h".} = int32

--- a/tests/nimony/ffi/deps/mimportcinttypedef.nim
+++ b/tests/nimony/ffi/deps/mimportcinttypedef.nim
@@ -1,0 +1,1 @@
+type CIntType* {.importc: "CIntType", header: "cinttype.h".} = int32

--- a/tests/nimony/ffi/timportcinttypedef.nim
+++ b/tests/nimony/ffi/timportcinttypedef.nim
@@ -1,0 +1,4 @@
+type CIntType {.importc: "CIntType", header: "cinttype.h".} = int32
+
+var x: CIntType
+discard x

--- a/tests/nimony/ffi/timportcinttypedef.nim
+++ b/tests/nimony/ffi/timportcinttypedef.nim
@@ -1,4 +1,4 @@
-type CIntType {.importc: "CIntType", header: "cinttype.h".} = int32
+type CIntTypeDef {.importc: "CIntTypeDef", header: "cinttypedef.h".} = int32
 
-var x: CIntType
+var x: CIntTypeDef
 discard x

--- a/tests/nimony/ffi/timportcinttypedefinmod.nim
+++ b/tests/nimony/ffi/timportcinttypedefinmod.nim
@@ -1,4 +1,4 @@
-import deps/mimportcinttype
+import deps/mimportcinttypedef
 
-var x: CIntType
+var x: CIntTypeDef
 discard x

--- a/tests/nimony/ffi/timportcinttypedefinmod.nim
+++ b/tests/nimony/ffi/timportcinttypedefinmod.nim
@@ -1,0 +1,4 @@
+import deps/mimportcinttype
+
+var x: CIntType
+discard x

